File: testkotlin/src/main/kotlin/ml/guxing/test/core/hibernate/models/EntityModel.kt
package ml.guxing.test.core.hibernate.models

import java.math.BigInteger
import javax.persistence.Id
import javax.persistence.MappedSuperclass
import javax.persistence.Version

@MappedSuperclass
open class EntityModel {

    @Id
    var id: BigInteger? = null
    @Version
    var version: Int? = null

}

File: testscala/src/main/scala/ml/guxing/test/core/test/scala/core/domain/model/EntityModel.scala
package ml.guxing.test.core.test.scala.core.domain.model

import java.math.BigInteger
import javax.persistence.{Id, MappedSuperclass, Version}

import scala.beans.BeanProperty

@MappedSuperclass
abstract class EntityModel {

  @Id
  @BeanProperty
  var id: BigInteger = null
  @Version
  @BeanProperty
  var version: Integer = null

}
